### ISSUE TITLE
Add Biometric / WebAuthn Enrollment and Authentication

### ISSUE THREAD

**Issue by binbashburns** [defenseunicorns/uds-identity-config#236]
### Describe the solution you'd like

- **Given** a user is setting up multi-factor authentication in Keycloak
- **When** the user reaches the MFA setup stage in the registration and authentication flows
- **Then** WebAuthn Passwordless Authentication (including biometric readers like TouchID, FaceID, Fingerprints, etc.) can be used for MFA

### Describe things you've considered
- The **_initial login_**/self-registration flows would need to be modified to include WebAuthn / Passwordless _**enrollment**_ (registration flow)
- Subsequent **_authentication_** flows (i.e. navigating to a mission app within UDS) would need to be updated to support WebAuthn / Biometric authentication (browser based authentication)

### Additional context
- [Medium Article](https://embesozzi.medium.com/keycloak-workshop-for-step-up-with-mfa-biometrics-authentication-passkeys-b7020ea9ae1b)
- [Keycloak Documentation](https://www.keycloak.org/docs/latest/server_admin/#webauthn_server_administration_guide)

#### Example Biometric Passkey Registration:
![](https://miro.medium.com/v2/resize:fit:640/format:webp/1*Sz44pozbA2FgOesUBepr4g.png)

![](https://miro.medium.com/v2/resize:fit:640/format:webp/1*xXbAZgSLfaZF2oxN2QYFZA.png)

#### Example Biometric Passkey Authentication:
![](https://miro.medium.com/v2/resize:fit:720/format:webp/1*Eclo_UUVVl0PytiSIiPGvw.png)

**Comment by UnicornChance** [comment]
### Problem:
Unable to configure authentication flows correctly for IdP registration/logins with webauthn unless we write a custom plugin for the `Post Login Flow`.

### Why:
Keycloak IdP's only have two flow overrides, called the `First login flow override` and `Post login flow`. The first login flow manages the first time a user sets up an account with the IdP. The Post Login Flow run after every successful login with an IdP. 

To configure keycloak to require users to attach a webauthn token, its easiest to set a `Required Action` so that all users are required to do so. However these required actions are not run until a users account has finished going through the other auth flows. In almost all cases this works as expected.

In the case of an IdP the `first login flow override` and the `post login flow` must complete before these required actions can run. This creates an issue, in order for the IdP to check for webauth tokens on logins in the future it needs to be added to the `post login flow`, however the first time a user registers with the IdP they never get the opportunity to setup the webauth token because its a required action and for their registration process to reach the required actions it needs to complete the `post login flow`, which requires that webauth token be setup.

tldr; There is a circular dependency for configuring a required webauth token on Keycloaks IdP's

### Solutions
* Custom plugin that can conditionally determine if this a new user registering or an old users authenticating. 
* Custom plugin that when called can force the required actions to run, adding this to the first login flow override would ensure that webauth token is already attached before reaching the `Post Login Flow`
* ???

### Other Considerations
Up till now we have taken the stance that when using an IdP, we are shifting all authentication to that Identity Provider. Meaning that this is a non-issue because it would be the providers responsibility to configure a webauth token or other OTP. i.e. Google's Idp that comes configured in uds realm, would need to manage the otp now keycloak.
